pkg/controller/directvolumemigration/rsync.go

```diff
@@ -79,623 +66,436 @@ const (
-		rsyncConf := rsyncConfig{
-			SshUser:   "root",
+		nnPair := cranemeta.NewNamespacedPair(
+			types.NamespacedName{Name: DirectVolumeMigrationRsyncTransfer, Namespace: srcNs},
+			types.NamespacedName{Name: DirectVolumeMigrationRsyncTransfer, Namespace: destNs},
+		)
+		endpoint, err := route_endpoint.GetEndpointFromKubeObjects(destClient, types.NamespacedName{
+			Name:      DirectVolumeMigrationRsyncTransferRoute,
 			Namespace: destNs,
-			PVCList:   pvcList,
-			Password:  password,
+		})
+		if err != nil {
+			return liberr.Wrap(err)
 		}
-		var tpl bytes.Buffer
-		temp, err := template.New("config").Parse(rsyncConfigTemplate)
+		stunnelTransport, err := stunnel_transport.GetTransportFromKubeObjects(
+			srcClient, destClient, nnPair, endpoint)
 		if err != nil {
-			return err
+			return liberr.Wrap(err)
 		}
-		err = temp.Execute(&tpl, rsyncConf)
+		pvcList, err := transfer.NewPVCPairList(pvcPairs...)
 		if err != nil {
-			return err
+			return liberr.Wrap(err)
 		}
-
-		configMap := corev1.ConfigMap{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: destNs,
-				Name:      DirectVolumeMigrationRsyncConfig,
-			},
+		labels := t.buildDVMLabels()
+		labels["purpose"] = DirectVolumeMigrationRsync
+		rsyncOptions, err := t.getRsyncTransferOptions()
+		if err != nil {
+			return liberr.Wrap(err)
 		}
-		configMap.Labels = t.Owner.GetCorrelationLabels()
-		configMap.Labels["app"] = DirectVolumeMigrationRsyncTransfer
-
-		err = yaml.Unmarshal(tpl.Bytes(), &configMap)
+		mutations, err := t.getRsyncContainerMutations(srcClient)
 		if err != nil {
-			return err
+			return liberr.Wrap(err)
 		}
-
-		// Create configmap on source + dest
-		// Note: when this configmap changes the rsync pod
-		// needs to restart
-		// Need to launch new pod when configmap changes
-		t.Log.Info("Creating Rsync Transfer Pod ConfigMap on destination cluster",
-			"configMap", path.Join(configMap.Namespace, configMap.Name))
-		err = destClient.Create(context.TODO(), &configMap)
-		if k8serror.IsAlreadyExists(err) {
-			t.Log.Info("Rsync Transfer Pod ConfigMap already exists on destination",
-				"configMap", path.Join(configMap.Namespace, configMap.Name))
-		} else if err != nil {
-			return err
+		rsyncOptions = append(rsyncOptions, mutations...)
+		rsyncOptions = append(rsyncOptions, rsync_transfer.WithDestinationPodLabels(labels))
+		transfer, err := rsync_transfer.NewTransfer(
+			stunnelTransport, endpoint, srcClient.RestConfig(), destClient.RestConfig(), pvcList, rsyncOptions...)
+		if err != nil {
+			return liberr.Wrap(err)
 		}
+		err = transfer.CreateServer(destClient)
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+	}
+	return nil
+}
 
-		// Before starting rsync transfer pod, must generate rsync password in a
-		// secret and pass it into the transfer pod
+func (t *Task) createRsyncTransferClients(srcClient compat.Client,
+	destClient compat.Client, nsMap map[string][]transfer.PVCPair) (rsyncClientOperationStatusList, error) {
```

Comment:
NIT: consider using pointers to reduce variable copies? `(*rsyncClientOperationStatusList, error)`